swift_targets.bzl

```diff
@@ -0,0 +1,233 @@
+def swift_get_compile_options(
+        exceptions = None,
+        rtti = None,
+        warning = None,
+        add_compile_options = None,
+        remove_compile_options = None,
+        cxx_standard = None,
+        c_standard = None):
+    if add_compile_options == None:
+        add_compile_options = []
+    if remove_compile_options == None:
+        remove_compile_options = []
+
+    for flag in add_compile_options + remove_compile_options:
+        if flag == "-fexceptions":
+            fail("Do not specify -fexceptions directly, use 'exceptions' argument instead")
+        if flag == "-fno-exceptions":
+            fail("Do not specify -fno-exceptions directly, avoid 'exceptions' argument instead")
+        if flag == "-frtti":
+            fail("Do not specify -frtti directly, use 'rtti' argument instead")
+        if flag == "-fno-rtti":
+            fail("Do not specify -fno-rtti directly, avoid 'rtti' argument instead")
+        if flag == "-Wno-error":
+            fail("Do not specify -Wno-error directly, use 'warning' to disable -Werror")
+
+    all_flags = []
+
+    if not warning:
+        all_flags += ["-Werror", "-Wno-error=deprecated-declarations"]
+
+    # '-Wall' '-Wunused-but-set-parameter' '-Wno-free-nonheap-object' flags are set by default by bazel
+    all_flags += [
+        "-Wcast-align",
+        "-Wcast-qual",
+        "-Wchar-subscripts",
+        "-Wcomment",
+        "-Wconversion",
+        "-Wdisabled-optimization",
+        "-Wextra",
+        "-Wfloat-equal",
+        "-Wformat",
+        "-Wformat-security",
+        "-Wformat-y2k",
+        "-Wimplicit-fallthrough",
+        "-Wimport",
+        "-Winit-self",
+        "-Winvalid-pch",
+        "-Wmissing-braces",
+        "-Wmissing-field-initializers",
+        # "-Wmissing-include-dirs",
```

Comment:
I commented out `-Wmissing-include-dirs` because gcc-11 (this is not the case with clang-14) complains when e.g. `include` directory "inherited" from the `swiftnav` library is not valid for the `ixcom` library.

I tried to workaround it with the `-I` flag but I couldn't find a solution. We should probably create a ticket about it.